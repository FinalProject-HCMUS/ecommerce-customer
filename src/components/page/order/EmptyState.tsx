import React from 'react';
import { Package } from 'lucide-react';

interface EmptyStateProps {
  searchTerm: string;
  onClearSearch?: () => void;
}

const EmptyState: React.FC<EmptyStateProps> = ({
  searchTerm,
  onClearSearch,
}) => {
  return (
    <div className="text-center py-12">
      <Package className="mx-auto h-12 w-12 text-gray-400" />
      <h3 className="mt-2 text-lg font-medium text-gray-900">
        No orders found
      </h3>
      <p className="mt-1 text-sm text-gray-500">
        {searchTerm
          ? "Try adjusting your search or filter to find what you're looking for."
          : "You haven't placed any orders yet."}
      </p>
      {searchTerm && onClearSearch && (
        <button
          onClick={onClearSearch}
          className="mt-4 px-4 py-2 bg-white border border-gray-300 rounded-md text-sm font-medium text-gray-700 hover:bg-gray-50"
        >
          Clear Search
        </button>
      )}
    </div>
  );
};

export default EmptyState;
