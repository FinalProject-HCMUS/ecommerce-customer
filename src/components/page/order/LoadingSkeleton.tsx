import React from 'react'

interface LoadingSkeletonProps {
  count?: number
}

const LoadingSkeleton: React.FC<LoadingSkeletonProps> = ({ count = 3 }) => {
  return (
    <div className="space-y-4">
      {Array.from({ length: count }).map((_, index) => (
        <div key={index} className="bg-white rounded-lg shadow-sm border border-gray-200 p-4 animate-pulse">
          <div className="flex items-center justify-between">
            <div className="flex items-center space-x-4">
              <div className="w-5 h-5 bg-gray-300 rounded-full"></div>
              <div>
                <div className="h-4 bg-gray-300 rounded w-32 mb-2"></div>
                <div className="h-3 bg-gray-200 rounded w-24"></div>
              </div>
            </div>
            <div className="flex items-center space-x-4">
              <div className="h-5 bg-gray-300 rounded w-20"></div>
              <div className="h-5 bg-gray-300 rounded w-16"></div>
              <div className="h-8 bg-gray-300 rounded w-24"></div>
              <div className="w-5 h-5 bg-gray-300 rounded-full"></div>
            </div>
          </div>
        </div>
      ))}
    </div>
  )
}

export default LoadingSkeleton
