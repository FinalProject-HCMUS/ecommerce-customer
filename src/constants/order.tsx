export const orderStatus = [
  { value: 'all', label: 'All Orders' },
  { value: 'NEW', label: 'New' },
  { value: 'PROCESSING', label: 'Processing' },
  { value: 'PACKAGED', label: 'Packaged' },
  { value: 'PICKED', label: 'Picked' },
  { value: 'SHIPPING', label: 'Shipping' },
  { value: 'DELIVERED', label: 'Delivered' },
  { value: 'CANCELLED', label: 'Cancelled' },
  { value: 'REFUNDED', label: 'Refunded' },
];
