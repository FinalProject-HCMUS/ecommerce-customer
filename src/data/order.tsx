import type { Order } from "../interfaces/temp/order"

// Mock data for demonstration
export const mockOrders: Order[] = [
  {
    id: "1",
    orderNumber: "ORD-2023-0001",
    date: "2023-04-15",
    status: "DELIVERED",
    items: [
      {
        id: "item1",
        name: "Premium Cotton T-Shirt",
        quantity: 2,
        price: 29.99,
        image: "/placeholder.svg?height=80&width=80",
      },
      {
        id: "item2",
        name: "Slim Fit Jeans",
        quantity: 1,
        price: 59.99,
        image: "/placeholder.svg?height=80&width=80",
      },
    ],
    total: 119.97,
    statusHistory: [
      { status: "NEW", date: "2023-04-15T08:30:00Z", note: "Order placed" },
      { status: "PROCESSING", date: "2023-04-15T09:15:00Z", note: "Payment confirmed" },
      { status: "PACKAGED", date: "2023-04-16T10:20:00Z", note: "Items packaged" },
      { status: "PICKED", date: "2023-04-16T14:45:00Z", note: "Ready for shipping" },
      { status: "SHIPPING", date: "2023-04-17T08:30:00Z", note: "In transit" },
      { status: "DELIVERED", date: "2023-04-20T13:25:00Z", note: "Package delivered" },
    ],
  },
  {
    id: "2",
    orderNumber: "ORD-2023-0002",
    date: "2023-05-20",
    status: "SHIPPING",
    items: [
      {
        id: "item3",
        name: "Hooded Sweatshirt",
        quantity: 1,
        price: 49.99,
        image: "/placeholder.svg?height=80&width=80",
      },
    ],
    total: 49.99,
    statusHistory: [
      { status: "NEW", date: "2023-05-20T10:15:00Z", note: "Order placed" },
      { status: "PROCESSING", date: "2023-05-20T11:30:00Z", note: "Payment confirmed" },
      { status: "PACKAGED", date: "2023-05-21T09:45:00Z", note: "Items packaged" },
      { status: "PICKED", date: "2023-05-21T15:20:00Z", note: "Ready for shipping" },
      { status: "SHIPPING", date: "2023-05-22T08:10:00Z", note: "In transit" },
    ],
  },
  {
    id: "3",
    orderNumber: "ORD-2023-0003",
    date: "2023-06-10",
    status: "PROCESSING",
    items: [
      {
        id: "item4",
        name: "Leather Jacket",
        quantity: 1,
        price: 199.99,
        image: "/placeholder.svg?height=80&width=80",
      },
      {
        id: "item5",
        name: "Wool Scarf",
        quantity: 1,
        price: 29.99,
        image: "/placeholder.svg?height=80&width=80",
      },
    ],
    total: 229.98,
    statusHistory: [
      { status: "NEW", date: "2023-06-10T14:25:00Z", note: "Order placed" },
      { status: "PROCESSING", date: "2023-06-10T15:40:00Z", note: "Payment confirmed" },
    ],
  },
  {
    id: "4",
    orderNumber: "ORD-2023-0004",
    date: "2023-07-05",
    status: "CANCELLED",
    items: [
      {
        id: "item6",
        name: "Summer Dress",
        quantity: 1,
        price: 79.99,
        image: "/placeholder.svg?height=80&width=80",
      },
    ],
    total: 79.99,
    statusHistory: [
      { status: "NEW", date: "2023-07-05T09:10:00Z", note: "Order placed" },
      { status: "PROCESSING", date: "2023-07-05T10:30:00Z", note: "Payment confirmed" },
      { status: "CANCELLED", date: "2023-07-06T11:15:00Z", note: "Cancelled by customer" },
    ],
  },
  {
    id: "5",
    orderNumber: "ORD-2023-0005",
    date: "2023-08-12",
    status: "REFUNDED",
    items: [
      {
        id: "item7",
        name: "Wireless Headphones",
        quantity: 1,
        price: 129.99,
        image: "/placeholder.svg?height=80&width=80",
      },
    ],
    total: 129.99,
    statusHistory: [
      { status: "NEW", date: "2023-08-12T13:20:00Z", note: "Order placed" },
      { status: "PROCESSING", date: "2023-08-12T14:15:00Z", note: "Payment confirmed" },
      { status: "PACKAGED", date: "2023-08-13T10:30:00Z", note: "Items packaged" },
      { status: "PICKED", date: "2023-08-13T15:45:00Z", note: "Ready for shipping" },
      { status: "SHIPPING", date: "2023-08-14T09:20:00Z", note: "In transit" },
      { status: "DELIVERED", date: "2023-08-17T14:10:00Z", note: "Package delivered" },
      { status: "REFUNDED", date: "2023-08-20T11:25:00Z", note: "Product returned and refunded" },
    ],
  },
]
