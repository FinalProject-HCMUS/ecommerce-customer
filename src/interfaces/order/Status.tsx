export enum Status {
  NEW = 'new',
  CANCELLED = 'cancelled',
  PROCESSING = 'processing',
  PACKAGED = 'packaged',
  PICKED = 'picked',
  SHIPPING = 'shipping',
  DELIVERED = 'delivered',
  REFUNDED = 'refunded',
}
