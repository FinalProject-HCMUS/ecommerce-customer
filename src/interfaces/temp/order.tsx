import type { ReactNode } from 'react';

// Order status types
export type OrderStatus =
  | 'NEW'
  | 'CANCELLED'
  | 'PROCESSING'
  | 'PACKAGED'
  | 'PICKED'
  | 'SHIPPING'
  | 'DELIVERED'
  | 'REFUNDED';

// Order item interface
export interface OrderItem {
  id: string;
  name: string;
  quantity: number;
  price: number;
  image: string;
}

// Status history entry
export interface StatusHistoryEntry {
  status: OrderStatus;
  date: string;
  note?: string;
}

// Order interface
export interface Order {
  id: string;
  orderNumber: string;
  date: string;
  status: OrderStatus;
  items: OrderItem[];
  total: number;
  statusHistory: StatusHistoryEntry[];
}

// Status configuration interface
export interface StatusConfig {
  icon: ReactNode;
  color: string;
  bgColor: string;
  borderColor: string;
  text: string;
}

// Status configuration record
export type StatusConfigRecord = Record<OrderStatus, StatusConfig>;
