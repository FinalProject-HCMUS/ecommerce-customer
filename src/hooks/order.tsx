import { useCallback, useEffect, useState } from 'react';
import { checkout, searchOrders } from '../services/apis/orderApis';
import { CheckoutRequest } from '../interfaces/order/CheckOut';
import { OrderResponse } from '../interfaces/order/OrderResponse';
import { CustomResponse } from '../interfaces/common/CustomResponse';
import { OrderSearchParams } from '../interfaces/order/OrderSearchParams';
import { Pageable } from '../interfaces';

export const useCheckout = () => {
  const [loading, setLoading] = useState(false);
  const [orderResponse, setOrderResponse] = useState<OrderResponse | null>(null);

  const performCheckout = async (checkoutData: CheckoutRequest): Promise<CustomResponse<OrderResponse> | undefined> => {
    setLoading(true);
    try {
      const response = await checkout(checkoutData);
      
      if (response.isSuccess && response.data) {
        setOrderResponse(response.data);
      }
      
      setLoading(false);
      return response;
    } catch {
      setLoading(false);
      return undefined;
    }
  };

  return {
    loading,
    orderResponse,
    performCheckout
  };
};

export const useOrderSearch = (initialParams?: OrderSearchParams) => {
  const [searchParams, setSearchParamsState] = useState<OrderSearchParams>(initialParams || {
    page: 0,
    size: 10,
  });
  const [orders, setOrders] = useState<OrderResponse[]>([]);
  const [loading, setLoading] = useState<boolean>(false);
  const [pageable, setPageable] = useState<Pageable<OrderResponse[]>>();

  const fetchOrders = useCallback(async () => {
    setLoading(true);
    console.log('Fetching orders with params:', searchParams);
    try {
      const response = await searchOrders(searchParams);
      if (response.isSuccess && response.data) {
        setOrders(response.data.content);
        setPageable(response.data);
      } 
    }  finally {
      setLoading(false);
    }
  }, [searchParams]);

  // Update search parameters and fetch results
  const setSearchParams = useCallback((params: Partial<OrderSearchParams>) => {
    setSearchParamsState(prev => {
      // If changing filters (not pagination), reset to first page
      if (params.keyword !== undefined || params.status !== undefined || params.size !== undefined) {
        return { ...prev, ...params, page: 0 };
      }
      return { ...prev, ...params };
    });
  }, []);

  // Function to manually refresh results
  const refresh = useCallback(async () => {
    await fetchOrders();
  }, [fetchOrders]);

  // Fetch orders whenever search params change
  useEffect(() => {
    fetchOrders();
  }, [fetchOrders]);

  return {
    orders,
    loading,
    pageable,
    searchParams,
    setSearchParams,
    refresh,
  };
};