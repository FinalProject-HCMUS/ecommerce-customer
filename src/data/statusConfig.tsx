import { Clock, CheckCircle, XCircle, Truck, Package } from 'lucide-react';
import type { ReactNode } from 'react';
import { Status } from '../interfaces/order/Status';

// Status configuration interface
export interface StatusConfig {
  icon: ReactNode;
  color: string;
  bgColor: string;
  borderColor: string;
  text: string;
}

// Status configuration record
export type StatusConfigRecord = Record<Status, StatusConfig>;

export const statusConfig: StatusConfigRecord = {
  [Status.NEW]: {
    icon: <Clock className="w-4 h-4 mr-1" />,
    color: 'text-purple-800',
    bgColor: 'bg-purple-100',
    borderColor: 'border-purple-200',
    text: 'New',
  },
  [Status.CANCELLED]: {
    icon: <XCircle className="w-4 h-4 mr-1" />,
    color: 'text-red-800',
    bgColor: 'bg-red-100',
    borderColor: 'border-red-200',
    text: 'Cancelled',
  },
  [Status.PROCESSING]: {
    icon: <Clock className="w-4 h-4 mr-1" />,
    color: 'text-amber-800',
    bgColor: 'bg-amber-100',
    borderColor: 'border-amber-200',
    text: 'Processing',
  },
  [Status.PACKAGED]: {
    icon: <Package className="w-4 h-4 mr-1" />,
    color: 'text-blue-800',
    bgColor: 'bg-blue-100',
    borderColor: 'border-blue-200',
    text: 'Packaged',
  },
  [Status.PICKED]: {
    icon: <CheckCircle className="w-4 h-4 mr-1" />,
    color: 'text-indigo-800',
    bgColor: 'bg-indigo-100',
    borderColor: 'border-indigo-200',
    text: 'Picked',
  },
  [Status.SHIPPING]: {
    icon: <Truck className="w-4 h-4 mr-1" />,
    color: 'text-cyan-800',
    bgColor: 'bg-cyan-100',
    borderColor: 'border-cyan-200',
    text: 'Shipping',
  },
  [Status.DELIVERED]: {
    icon: <CheckCircle className="w-4 h-4 mr-1" />,
    color: 'text-green-800',
    bgColor: 'bg-green-100',
    borderColor: 'border-green-200',
    text: 'Delivered',
  },
  [Status.REFUNDED]: {
    icon: <CheckCircle className="w-4 h-4 mr-1" />,
    color: 'text-rose-800',
    bgColor: 'bg-rose-100',
    borderColor: 'border-rose-200',
    text: 'Refunded',
  },
};