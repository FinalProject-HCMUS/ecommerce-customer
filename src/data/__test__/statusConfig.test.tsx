import { statusConfig } from '../statusConfig';
import { Status } from '../../interfaces/order/Status';

describe('statusConfig', () => {
  test('should export an object with all Status keys', () => {
    expect(typeof statusConfig).toBe('object');
    const expectedStatuses = [
      Status.NEW,
      Status.CANCELLED,
      Status.PROCESSING,
      Status.PACKAGED,
      Status.PICKED,
      Status.SHIPPING,
      Status.DELIVERED,
      Status.REFUNDED,
    ];
    expectedStatuses.forEach(status => {
      expect(statusConfig).toHaveProperty(status);
    });
  });

  test('each status config should have correct structure', () => {
    Object.values(statusConfig).forEach(cfg => {
      expect(cfg).toHaveProperty('icon');
      expect(cfg).toHaveProperty('color');
      expect(cfg).toHaveProperty('bgColor');
      expect(cfg).toHaveProperty('borderColor');
      expect(cfg).toHaveProperty('text');
      expect(typeof cfg.color).toBe('string');
      expect(typeof cfg.bgColor).toBe('string');
      expect(typeof cfg.borderColor).toBe('string');
      expect(typeof cfg.text).toBe('string');
    });
  });

  test('should have correct text for each status', () => {
    expect(statusConfig[Status.NEW].text).toBe('New');
    expect(statusConfig[Status.CANCELLED].text).toBe('Cancelled');
    expect(statusConfig[Status.PROCESSING].text).toBe('Processing');
    expect(statusConfig[Status.PACKAGED].text).toBe('Packaged');
    expect(statusConfig[Status.PICKED].text).toBe('Picked');
    expect(statusConfig[Status.SHIPPING].text).toBe('Shipping');
    expect(statusConfig[Status.DELIVERED].text).toBe('Delivered');
    expect(statusConfig[Status.REFUNDED].text).toBe('Refunded');
  });

  test('should match snapshot', () => {
    expect(statusConfig).toMatchSnapshot();
  });
});