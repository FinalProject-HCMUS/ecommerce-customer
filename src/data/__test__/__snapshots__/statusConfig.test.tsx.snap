// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`statusConfig should match snapshot 1`] = `
{
  "CANCELLED": {
    "bgColor": "bg-red-100",
    "borderColor": "border-red-200",
    "color": "text-red-800",
    "icon": <CircleX
      className="w-4 h-4 mr-1"
    />,
    "text": "Cancelled",
  },
  "DELIVERED": {
    "bgColor": "bg-green-100",
    "borderColor": "border-green-200",
    "color": "text-green-800",
    "icon": <CircleCheckBig
      className="w-4 h-4 mr-1"
    />,
    "text": "Delivered",
  },
  "NEW": {
    "bgColor": "bg-purple-100",
    "borderColor": "border-purple-200",
    "color": "text-purple-800",
    "icon": <Clock
      className="w-4 h-4 mr-1"
    />,
    "text": "New",
  },
  "PACKAGED": {
    "bgColor": "bg-blue-100",
    "borderColor": "border-blue-200",
    "color": "text-blue-800",
    "icon": <Package
      className="w-4 h-4 mr-1"
    />,
    "text": "Packaged",
  },
  "PICKED": {
    "bgColor": "bg-indigo-100",
    "borderColor": "border-indigo-200",
    "color": "text-indigo-800",
    "icon": <CircleCheckBig
      className="w-4 h-4 mr-1"
    />,
    "text": "Picked",
  },
  "PROCESSING": {
    "bgColor": "bg-amber-100",
    "borderColor": "border-amber-200",
    "color": "text-amber-800",
    "icon": <Clock
      className="w-4 h-4 mr-1"
    />,
    "text": "Processing",
  },
  "REFUNDED": {
    "bgColor": "bg-rose-100",
    "borderColor": "border-rose-200",
    "color": "text-rose-800",
    "icon": <CircleCheckBig
      className="w-4 h-4 mr-1"
    />,
    "text": "Refunded",
  },
  "SHIPPING": {
    "bgColor": "bg-cyan-100",
    "borderColor": "border-cyan-200",
    "color": "text-cyan-800",
    "icon": <Truck
      className="w-4 h-4 mr-1"
    />,
    "text": "Shipping",
  },
}
`;
